sysfs-led-mind2-v1.c

```diff
@@ -0,0 +1,355 @@
+        if( !led_state_mind2v1_brobe(&self->led[idx], &paths->led[idx]) )
+            goto cleanup;
+
+    res = true;
+
+cleanup:
+    if( !res )
+        leds_state_mind2v1_close(self);
+
+    return res;
+}
+
+static void
+leds_state_mind2v1_update_power(leds_state_mind2v1_t *self)
+{
+    bool power = false;
+
+    for( int idx = 0; idx < MIND2V1_LED_COUNT; ++idx )
+        if( (power = led_state_mind2v1_is_active(&self->led[idx])) )
+            break;
+
+    sysfsval_set(self->cached_power, power);
+}
+
+static void
+leds_state_mind2v1_set_value(leds_state_mind2v1_t *self, int idx, int r, int g, int b)
+{
+    if( leds_state_mind2v1_valid_index(idx) )
+        led_state_mind2v1_set_value(&self->led[idx], r, g, b);
+}
+
+/* ========================================================================= *
+ * LED_CONTROL_MIND2V1
+ * ========================================================================= */
+
+static void
+led_control_mind2v1_value_cb(void *data, int r, int g, int b)
+{
+    leds_state_mind2v1_t *state = data;
+
+    leds_state_mind2v1_set_value(state, MIND2V1_LED_INNER, r, g, b);
+#if DIFFERENTIATE_OUTER_LED
+    leds_state_mind2v1_set_value(state, MIND2V1_LED_OUTER, g, b, r);
+#else
+    leds_state_mind2v1_set_value(state, MIND2V1_LED_OUTER, r, g, b);
+#endif
+    leds_state_mind2v1_update_power(state);
+}
+
+static void
+led_control_mind2v1_close_cb(void *data)
+{
+    leds_state_mind2v1_t *state = data;
+
+    leds_state_mind2v1_quit(state);
+}
+
+static bool
+led_control_mind2v1_static_probe(leds_state_mind2v1_t *state)
+{
+    static const leds_paths_mind2v1_t paths = {
+        .power = "/sys/class/leds/Power/brightness",
+        .led   = {
+            {
+                .brightness = "/sys/class/leds/Irgb/brightness",
+                .red        = "/sys/class/leds/Ired/brightness",
+                .green      = "/sys/class/leds/Igreen/brightness",
+                .blue       = "/sys/class/leds/Iblue/brightness",
+            },
+            {
+                .brightness = "/sys/class/leds/Orgb/brightness",
+                .red        = "/sys/class/leds/Ored/brightness",
+                .green      = "/sys/class/leds/Ogreen/brightness",
+                .blue       = "/sys/class/leds/Oblue/brightness",
+            }
+        }
+    };
+
+    return leds_state_mind2v1_probe(state, &paths);
+}
+
+static bool
+led_control_mind2v1_dynamic_probe(leds_state_mind2v1_t *state)
+{
+    // XXX: No configuration tweaks for now
+    (void)state;
+    return false;
+}
+
+bool
+led_control_mind2v1_probe(led_control_t *self)
+{
+    static leds_state_mind2v1_t state = { };
+
+    bool res = false;
+
+    leds_state_mind2v1_init(&state);
+
+    self->name   = "mind2v1";
+    self->data   = &state;
+    self->enable = NULL;
+    self->blink  = NULL;
+    self->value  = led_control_mind2v1_value_cb;
+    self->close  = led_control_mind2v1_close_cb;
+
+    // XXX: Disable breathing for now
```

Comment:
should this (`XXX`)be TODO or FIXME?

Reply:
It is one, another, both or neither of those depending on future needs - so, a kind of general heads up.